lib/rubocop/cop/lint/ordered_magic_comments.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      #
+      # This cop checks coding magic comment should be first line,
+      # or next of shebang.
+      #
+      # @example
+      #   # bad
+      #
+      #   # frozen_string_literal: true
+      #   # coding: ascii
+      #   p [''.frozen?, ''.encoding] #=> [true, #<Encoding:UTF-8>]
+      #
+      #   # good
+      #
+      #   # coding: ascii
+      #   # frozen_string_literal: true
+      #   p [''.frozen?, ''.encoding] #=> [true, #<Encoding:US-ASCII>]
+      #
+      #   # good
+      #
+      #   #!/usr/bin/env ruby
+      #   # coding: ascii
+      #   # frozen_string_literal: true
+      #   p [''.frozen?, ''.encoding] #=> [true, #<Encoding:US-ASCII>]
+      #
+      class OrderedMagicComments < Cop
+        include FrozenStringLiteral
+
+        MSG = 'Use coding magic comment should be first line, ' \
```

Comment:
I guess the message should just "The `coding` magic comment should precede all other magic comments." or something like this.